### PR TITLE
2 replicate fab results, update model evaluation

### PR DESCRIPTION
**Pull request by lollcat**
Solves #7 
- Got the 2D Many Well problem, GMM problem (nice quick to run examples)
- 16D Many Well problem working (takes longer to run - so recommend using GPU / colab)
- Updated the evaluation loop for calculation of ESS with a larger number of samples (intermittently during training), and target specific metrics (e.g. mean test set log prob). 
- Updated README.md with some example plots from the examples (these plots are still a bit rough, but will leave cleaning them up for a later issue). 
-  Examples working with float32 precision
- Update a line in the HMC code that was causing lots of Nan print messages during AIS.

### CODE REVIEW

**Comment by VincentStimper** [comment]
Also replace [this](https://github.com/lollcat/FAB-TORCH/blob/38d165fa6e3de03012751e3c5cd35cb7168ee0a9/examples/make_flow/make_realnvp_normflow.py#L20) by `nf.flows.InvertibleAffine(dim)` since it is much more expressive. It does not work in the current version of `torch` due to a bug in the matrix inverse function, but it works with previous versions such as the LTS version (1.8.2), see [here](https://pytorch.org/get-started/locally/).

**Comment by VincentStimper** [comment]
I think you uploaded all the output of the many well problem again with your last commit. Can you remove it, please? Just do _Remove all output_ before committing it.
